frontend/src/pages/listing/SellListing.tsx

```diff
@@ -0,0 +1,271 @@
+import { useEffect, useState } from 'react';
+import { Link, useLoaderData, useParams, useNavigate } from 'react-router-dom';
+import {
+  CardDetailsLoaderData,
+  YugiohCardListing,
+  YugiohCardSellListing,
+} from '../../services/yugioh/types';
+import { yugiohService } from '../../services/yugioh/yugiohService';
+import { useEffectAfterInitialLoad } from '../../util/useEffectAfterInitialLoad';
+import YugiohCardMarket from '../../components/yugioh/table/market/YugiohCardMarket';
+import { Pagination } from '@mui/material';
+import ListingTopBar from '../../components/sellListing/ListingTopBar';
+import NewListingTopBar from '../../components/sellListing/NewListingTopBar';
+import PaymentsIcon from '@mui/icons-material/Payments';
+import DeleteIcon from '@mui/icons-material/Delete';
+import DiamondIcon from '@mui/icons-material/Diamond';
+import TagIcon from '@mui/icons-material/Tag';
+import VisibilityIcon from '@mui/icons-material/Visibility';
+
+function SellListing(): JSX.Element {
+  const [page, setPage] = useState(1);
+  const data = useLoaderData() as CardDetailsLoaderData;
+  const params = useParams();
+  const id = Number(params.id);
+  const cardId = Number(params.cardid);
+  const { cardInSet, cardListings: cardListingsData } = data;
+  const [cardListings, setCardListings] = useState(cardListingsData);
+  const navigate = useNavigate();
+
+  const [formData, setFormData] = useState<YugiohCardListing>({
+    id: 0,
+    quantity: 0,
+    condition: 'poor',
+    price: 0,
+    is_sold: true,
+    is_listed: true,
+    card: 0,
+    user_name: '',
+    user: 0,
+    card_name: '',
+    card_set_id: 0,
+  });
+
+  useEffect(() => {
+    async function loadCardListing() {
+      const currentCard: YugiohCardListing = await yugiohService.getListingById(Number(params.id));
+      if (currentCard && currentCard !== null) {
+        setFormData(() => ({
```

Comment:
Use destructuring, instead of manual prop assignment. If there were 120 props, would you go for manually assigning them all?